src/main/java/rx/Notification.java

```diff
@@ -158,15 +158,24 @@ public boolean isOnNext() {
      * @param observer the target observer to call onXXX methods on based on the kind of this Notification instance
      */
     public void accept(Observer<? super T> observer) {
-        if (isOnNext()) {
+        switch (kind) {
+        case OnNext:
             observer.onNext(getValue());
-        } else if (isOnCompleted()) {
-            observer.onCompleted();
-        } else if (isOnError()) {
+            break;
+        case OnError:
             observer.onError(getThrowable());
+            break;
+        case OnCompleted:
+            observer.onCompleted();
+            break;
+        default:
+            break;
```

Comment:
not sure your thoughts, but for exhaustive enum switches I prefer to `throw new AssertionError()` in the default case since it would be a programmer error for this branch to every be taken.